android/src/main/java/com/reactnativesimcardsmanager/SimCardsManagerModule.java

```diff
@@ -125,25 +125,12 @@ private void handleResolvableError(Promise promise, Intent intent) {
     try {
       // Resolvable error, attempt to resolve it by a user action
       // FIXME: review logic of resolve functions
-      PendingIntent callbackIntent = PendingIntent.getBroadcast(mReactContext, 3,
+      int resolutionRequestCode = 3;
+      PendingIntent callbackIntent = PendingIntent.getBroadcast(mReactContext, resolutionRequestCode,
           new Intent(ACTION_DOWNLOAD_SUBSCRIPTION), PendingIntent.FLAG_UPDATE_CURRENT |
               PendingIntent.FLAG_MUTABLE);
 
-      int resolutionRequestCode = 3;
       mgr.startResolutionActivity(mReactContext.getCurrentActivity(), resolutionRequestCode, intent, callbackIntent);
-      ActivityEventListener activityEventListener = new BaseActivityEventListener() {
-        @Override
-        public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent data) {
-          if (requestCode == resolutionRequestCode) {
```

Comment:
https://forum.developer.samsung.com/t/esim-download-does-not-work-on-galaxy-s20-since-android-11/10343